MvvmCross/Binding/Extensions/MvxEnumerableExtensions.cs

```diff
@@ -84,5 +86,25 @@ public static object ElementAt(this IEnumerable items, int position)
 
             return enumerator.Current;
         }
+
+        public static IEnumerable Filter(this IEnumerable items, Func<object, bool> predicate)
+        {
+            if (items == null)
+                return null;
+
+            var enumerator = items.GetEnumerator();
```

Comment:
Whilst perhaps not as efficient from a performance perspective, a simple Where linq clause would make this easier to follow?

Reply:
Linq queries do not work on non-generic IEnumerable, that is why we have these extensions in the first place.

Reply:
doh! yeh of course, it requires generics. We should be able to at least use a basic for loop here, right?